### PR TITLE
ng_netreg: warn about context overwrite

### PR DESCRIPTION
**Pull request by authmillenon**

### CODE REVIEW

File: sys/include/net/ng_netreg.h [hunk 1]
```diff
@@ -75,6 +75,9 @@ void ng_netreg_init(void);
  * @param[in] entry     An entry you want to add to the registry with
  *                      ng_netreg_entry_t::pid and ng_netreg_entry_t::demux_ctx set.
  *
+ * @warning Call ng_netreg_unregister() *before* you leave the context you
+ *          allocated @p entry in. Otherwise it might get overwritten.
+ *
```
Comment (PeterKietzmann) [inline sys/include/net/ng_netreg.h]: Sorry if this is a stupid question but shouldn't this warning be part of the documentation for `ng_netreg_unregister()`?

Comment (authmillenon) [inline sys/include/net/ng_netreg.h]: But in the hackathon it wasn't clear for non-RIOTers that calling `ng_netreg_register()` on a stack-allocated object is not a good idea, if you don't unregister it before popping the current stack frame.

Comment (kaspar030) [inline sys/include/net/ng_netreg.h]: > it wasn't clear for non-RIOTers

not only to non-RIOTers. ;)

**Comment by PeterKietzmann** [comment]
Ok, seems that we can live with it. ACK

**Comment by OlegHahm** [comment]
Travis fails because of shut down embdebian repository. Rebase would help, but since this is documentation only, we could go without CI checks. @kaspar030, are you satisfied with the documentation now?

**Pull request merged**